addOns/llm/src/main/java/org/zaproxy/addon/llm/ui/ImportDialog.java

```diff
@@ -0,0 +1,315 @@
+        fieldsPanel.add(
+                getChooseFileButton(),
+                LayoutHelper.getGBC(2, fieldsRow, 1, 0.5, new Insets(0, 4, 4, 0)));
+
+        int row = 0;
+        add(fieldsPanel, LayoutHelper.getGBC(0, row, 2, 1.0, new Insets(8, 8, 4, 8)));
+        row++;
+        var requiredFieldsLabel =
+                new ZapHtmlLabel(
+                        "<html><font color=red>*</font> "
+                                + Constant.messages.getString("llm.importDialog.requiredFields")
+                                + "</html>");
+        Font font = requiredFieldsLabel.getFont();
+        requiredFieldsLabel.setFont(FontUtils.getFont(font, FontUtils.Size.much_smaller));
+        requiredFieldsLabel.setHorizontalAlignment(JLabel.RIGHT);
+        add(requiredFieldsLabel, LayoutHelper.getGBC(0, row, 2, 1.0, new Insets(4, 8, 4, 8)));
+        row++;
+        add(getCancelButton(), LayoutHelper.getGBC(0, row, 1, 0.5, new Insets(4, 8, 8, 4)));
+        add(getImportButton(), LayoutHelper.getGBC(1, row, 1, 0.5, new Insets(4, 4, 8, 8)));
+        row++;
+        add(getProgressBar(), LayoutHelper.getGBC(0, row, 2, 1.0, new Insets(0, 8, 8, 8)));
+        pack();
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+    }
+
+    private boolean importSwagger()
+            throws IOException, URISyntaxException, ApiException, DatabaseException {
+
+        String swaggerLocation = getSwaggerField().getText();
+        LlmOptionsParam llmOptionsParam = extLlm.getOptionsParam();
+        Integer endpointCount = 0;
+
+        if (StringUtils.isEmpty(llmOptionsParam.getApiKey())) {
+            showWarningDialog(Constant.messages.getString("llm.options.apikey.error.undefinded"));
+            throw new RuntimeException(
+                    Constant.messages.getString("llm.options.apikey.error.undefinded"));
+        }
+
+        if (StringUtils.isEmpty(llmOptionsParam.getEndpoint())) {
+            showWarningDialog(Constant.messages.getString("llm.options.endpoint.error.undefinded"));
+            throw new RuntimeException(
+                    Constant.messages.getString("llm.options.endpoint.error.undefinded"));
+        }
+
+        LlmCommunicationService llmCommunicationService =
+                new LlmCommunicationService(
+                        llmOptionsParam.getModelName(), llmOptionsParam.getApiKey(), llmOptionsParam.getEndpoint());
+
+        if (StringUtils.isEmpty(swaggerLocation)) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        "llm.importDialog.error.missingSwagger"));
+                        getSwaggerField().requestFocusInWindow();
+                    });
+            return false;
+        }
+
+        try {
+            new URL(swaggerLocation).toURI();
+            new URI(swaggerLocation, true);
+            // implement logic here
```

Comment:
Is something still outstanding here?

If so it should be dealt with, if not then remove the comment.